org.mwc.debrief.legacy/src/Debrief/ReaderWriter/Replay/ImportReplay.java

```diff
@@ -1255,32 +1285,52 @@ final private String getThisSymProperty(final String symbology,
     return elementsAdded;
   }
   
-  public final List<Editable> importThis(final String text,Layers existingLayers) {
-    List<Editable> elementsAdded = new ArrayList<>();
+  public final void importThis(final String text) {
     String[] lines = text.split("\\r?\\n");
     for(String line:lines) {
       if(!line.trim().isEmpty()) {
         readLine(line.trim());
       }
     }
-    Layers tempLayers = getLayers();
-    Enumeration<Editable> tempElements = tempLayers.elements();
+  }
+  
+  public final void injectContent(Layers from,Layers destination,boolean performAdd) {
+    Enumeration<Editable> tempElements = from.elements();
     //now add to the plot's layers object
     while(tempElements.hasMoreElements()) {
       Editable l = tempElements.nextElement();
-      Layer existingLayer = existingLayers.findLayer(l.getName());
+      Layer existingLayer = destination.findLayer(l.getName());
       if(existingLayer==null) {
         Layer layerToAdd = (Layer)l;
-        existingLayers.addThisLayer(layerToAdd);
-        elementsAdded.add(layerToAdd);
+        if(performAdd) {
+          destination.addThisLayer(layerToAdd);
+        }
+        else {
+          destination.removeThisEditable(null, l);
```

Comment:
@ayeshare - I'm pretty sure that `destination.removeThisLayer(layerToAdd);` would be a better way of deleting hte layer.

Reply:
done